packages/quantic/scripts/npm/utils.js

```diff
@@ -0,0 +1,36 @@
+const pathlib = require('path');
+
+function quanticIsDependency() {
+  try {
+    // path to npm scripts is expected to be: project_name/node_modules/@coveo/quantic/scripts/npm
+    const upperDirectories = __dirname.split(pathlib.sep);
+    const dirCountAfterNodeModulesParent = 5;
+    const nodeModulesParent = upperDirectories.slice(
+      0,
+      -dirCountAfterNodeModulesParent
+    );
+    require.resolve(
```

Comment:
:thinking: Experiencing an error inside the `require.resolve()` invokation would simply return `false` and make other functions just skip over the installation process silently.

We should at least log the error in the `catch` block to help troubleshooting.

Reply:
Experiencing an error inside the require.resolve() invocation would mean that we are not in the case where quantic is installed as a dependency but in the case where we just did run the `npm install `command inside the Quantic repo cause doing this also triggers the pre/post install scripts. That's why I didn't log an error cause it's not really an error.
Do you think that we still need to log the error or a message in that case?

Reply:
Well, you are assuming that the only reason this invocation would fail is if Quantic is not used as a dependency. What if the invocation fails for an unrelated reason like a permission or memory issue?

One way to do it would be to silence the expected error, and log any unexpected error.